src/AstTranslator.cpp

```diff
@@ -303,40 +303,40 @@ std::unique_ptr<RamCondition> AstTranslator::translateConstraint(
             for (size_t i = 0; i < arity; i++) {
                 values.push_back(translator.translateValue(args[i], index));
             }
-            for (size_t i = 0; i < auxiliaryArity; i++) {
+            // we don't care about the provenance columns when doing the existence check
+            if (Global::config().has("provenance")) {
+                // undefined value for rule number
                 values.push_back(std::make_unique<RamUndefValue>());
+                // add the height annotation for provenanceNotExists
+                for (size_t h = 1; h < auxiliaryArity; h++) {
```

Comment:
If `h` is short for `height`, then use `height`. Otherwise use the 'standard' meaningless iterator variable `j`.